sync/work/src/main/java/com/google/samples/apps/nowinandroid/sync/services/SyncNotificationsService.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.sync.services
+
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.google.samples.apps.nowinandroid.core.data.util.SyncManager
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+private const val SYNC_TOPIC = "sync"
+
+@AndroidEntryPoint
+class SyncNotificationsService : FirebaseMessagingService() {
+
+    @Inject
+    lateinit var syncManager: SyncManager
+
+    override fun onMessageReceived(message: RemoteMessage) {
+        if (SYNC_TOPIC == message.from) {
+            syncManager.requestSync()
```

Comment:
So instead of periodically polling the backend to see if there are new updates, a message from firebase triggers a sync?

Reply:
Yes exactly.

Reply:
@tunjid I like it, much better solution imo. Two questions:
1. Is the backend part done, does it work on staging and production?
2. Do we need WorkManager at all in that case?

Reply:
1. Backend isn't done, I'll send you the design doc.
2. We still need `WorkManager` to sync on start up and we also also have foreground execution time limits from when we receive the FCM message so WorkManager is still our best bet